### PR TITLE
add rsync include,exclude support

### PR DESCRIPTION
**Pull request by juanvallejo**
Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1559693

Adds support for `--include` and `--exclude` flags with the tar strategy.

cc @soltysh @smarterclayton

### CODE REVIEW

File: pkg/oc/cli/cmd/rsync/util.go [hunk 1]
```diff
@@ -84,14 +84,26 @@ func rsyncFlagsFromOptions(o *RsyncOptions) []string {
...
-func rsyncSpecificFlags(o *RsyncOptions) []string {
+func tarFlagsFromOptions(o *RsyncOptions) []string {
 	flags := []string{}
+	if !o.Quiet {
+		flags = append(flags, "-v")
+	}
 	if len(o.RsyncInclude) > 0 {
-		flags = append(flags, "--include")
+		for _, include := range o.RsyncInclude {
+			flags = append(flags, fmt.Sprintf("**/%s", include))
```
Comment (smarterclayton) [inline pkg/oc/cli/cmd/rsync/util.go]: does this make the behavior consistent with rsync, or with our doc?

Comment (juanvallejo) [reply pkg/oc/cli/cmd/rsync/util.go]: @smarterclayton 
Will update to match rsync behavior.

From local testing, the current include / exclude rsync behavior is as follows:

```
$ oc rsync --include="*.yaml" --exclude="ignore.*" mypod-1-x29xk:/tmp/myplugin ./
receiving incremental file list
myplugin/
myplugin/LICENSE
myplugin/ignore.yaml
myplugin/plugin.yaml
```

Wanted to double-check if this was behaving as we expect? (I would have expected `myplugin/ignore.yaml` to not have been included)

File: pkg/oc/cli/cmd/rsync/copytar.go [hunk 2]
```diff
@@ -203,13 +209,30 @@ func (r *tarStrategy) String() string {
...
+	exclude := []string{}
+	for _, pattern := range excludes {
+		exclude = append(exclude, fmt.Sprintf("--exclude=%s", pattern))
+	}
+
 	var cmd []string
 	if strings.HasSuffix(sourceDir, "/") {
-		cmd = []string{"tar", "-C", sourceDir, "-c", "."}
+		include := []string{"."}
+		include = append(include, includes...)
```
Comment (juanvallejo) [inline pkg/oc/cli/cmd/rsync/copytar.go]: @smarterclayton @soltysh in both of these branches I attempt to mimic the behavior of the `rsync` strategy when `--include` is provided, however this does not seem to work with wildcards / patterns. Any time I provide an `--include` when copying *from* a pod *to* my local env, I receive an error [1] unless I specify an exact filename.

Per the `tar` documentation, there is a `--wildcard` flag I can use, but I seem to get the same error regardless.

1.
```
$ oc rsync --strategy=tar --include="*.yaml" --exclude=LICENSE mypod-1-x29xk:/tmp/myplugin/ ./
tar: *.yaml: Cannot stat: No such file or directory
tar: Exiting with failure status due to previous errors
error: error creating remote tar of source directory: command terminated with exit code 2
```

Running the same command above with `loglevel 4` gives me the command being executed as follows:
```
Remote tar command: tar -C /tmp/myplugin/ -c . *.yaml --exclude=LICENSE
```

Comment (juanvallejo) [reply pkg/oc/cli/cmd/rsync/copytar.go]: cc @soltysh ^

Comment (soltysh) [reply pkg/oc/cli/cmd/rsync/copytar.go]: @smarterclayton ^^

**Comment by juanvallejo** [comment]
@smarterclayton friendly ping

File: pkg/oc/cli/cmd/rsync/copytar.go (hunk 2 above)
Comment (juanvallejo) [reply pkg/oc/cli/cmd/rsync/copytar.go]: @smarterclayton friendly ping

**Pull request merged**